lxc/file.go

```diff
@@ -264,6 +265,12 @@ func (c *cmdFilePull) Run(cmd *cobra.Command, args []string) error {
 			return err
 		}
 		targetIsDir = true
+	} else if len(targetDirs) > 1 && c.file.flagMkdir {
+		dirs := strings.Join(targetDirs[:len(targetDirs)-1], string(os.PathSeparator))
+		err := os.MkdirAll(dirs, 0755)
```

Comment:
If the goal is to get the parent directory of the target, then `filepath.Base` should do that for you.